packages/text-field/Input.tsx

```diff
@@ -121,7 +121,7 @@ export default class Input<T extends HTMLElement = HTMLInputElement> extends Rea
 
     this.handleValidationAttributeUpdate(prevProps);
 
-    if (disabled !== prevProps.disabled) {
+    if ((disabled !== prevProps.disabled || prevProps.foundation !== foundation) && foundation) {
```

Comment:
what if we change this from `...prevProps.foundation !== foundation) && foundation` to `!prevProps.foundation && foundation`? Seems clearer. If that works can you make that a variable and use it on line 133 as well

Reply:
I extracted this checking to a variable, I think it's better now